boring/src/ssl/mod.rs

```diff
@@ -641,6 +641,42 @@ impl SslCurve {
 
     #[cfg(feature = "pq-experimental")]
     pub const P256_KYBER768_DRAFT00: SslCurve = SslCurve(ffi::NID_P256Kyber768Draft00);
+
+    cfg_if::cfg_if! {
+        if #[cfg(feature = "pq-preferred")] {
+            const DEFAULT_CURVES: &[SslCurve] = &[
+                #[cfg(not(feature = "fips"))]
+                #[cfg(not(feature = "fips-link-precompiled"))]
+                SslCurve::X25519_KYBER512_DRAFT00,
+                SslCurve::P256_KYBER768_DRAFT00,
+                #[cfg(not(feature = "fips"))]
+                #[cfg(not(feature = "fips-link-precompiled"))]
+                SslCurve::X25519,
+                SslCurve::SECP256R1,
+                SslCurve::SECP384R1,
+            ];
+        } else if #[cfg(feature = "pq-supported")] {
+            const DEFAULT_CURVES: &[SslCurve] = &[
+                #[cfg(not(feature = "fips"))]
+                #[cfg(not(feature = "fips-link-precompiled"))]
+                SslCurve::X25519,
+                SslCurve::SECP256R1,
+                SslCurve::SECP384R1,
+                #[cfg(not(feature = "fips"))]
+                #[cfg(not(feature = "fips-link-precompiled"))]
+                SslCurve::X25519_KYBER512_DRAFT00,
+                SslCurve::P256_KYBER768_DRAFT00,
+            ];
+        } else {
+            const DEFAULT_CURVES: &[SslCurve] = &[
+                #[cfg(not(feature = "fips"))]
+                #[cfg(not(feature = "fips-link-precompiled"))]
+                SslCurve::X25519,
+                SslCurve::SECP256R1,
+                SslCurve::SECP384R1,
+            ];
```

Comment:
To not introduce breaking changes, let's set the default curves only if any of the features is enabled

Reply:
That makes sense, but one of the goals of this PR is to disable X25519 if the `fips` flag is set. Do you agree with this goal?

Another way to achieve the same effect would be to call `SSL_CTX_set_compliance_policy()`, which configures P256 and P384 only: https://boringssl.googlesource.com/boringssl/+/refs/heads/master/include/openssl/ssl.h#5326. If you agree with this goal, then let's do this instead?

Reply:
It turns out that some of the tests related to SRTP fail if this compliance policy is set. I've fallen back to just explicitly enabling FIPS only key exchanges.